## Imports

In [ ]:
import pandas as pd

In [61]:
import matplotlib.pyplot as plt

In [62]:
import seaborn as sns

## Variables

In [34]:
dir1 = "/home/aweidman/data/bills_1Nov2023/"

In [46]:
csv_bills = dir1+"bills.csv"

In [47]:
csv_people = dir1+"people.csv"

In [48]:
csv_sponsors = dir1+"sponsors.csv"

In [49]:
csv_history = dir1+"history.csv"

In [50]:
csv_votes = dir1+"votes.csv"

In [51]:
csv_documents = dir1+"documents.csv"

## Dataframe1: Bills

In [52]:
df_bills = pd.read_csv(csv_bills)

In [31]:
df_bills.head(2)

bill_id  session_id bill_number  status status_desc status_date   
0  1741372        2041         HB1       1  Introduced  2023-03-14  \
1  1760974        2041         HB2       2   Engrossed  2023-05-15   

                                               title   
0  Water Quality Certification and Energy Project...  \
1                      Secure the Border Act of 2023   

                                         description  committee_id committee   
0  To lower energy costs by increasing American e...             0       NaN  \
1  To secure the borders of the United States, an...             0       NaN   

  last_action_date                                        last_action   
0       2023-03-30  The Clerk was authorized to correct section nu...  \
1       2023-05-16  Read the second time. Placed on Senate Legisla...   

                                     url   
0  https://legiscan.com/US/bill/HB1/2023  \
1  https://legiscan.com/US/bill/HB2/2023   

                                          state_link  
0  https://www.congress.gov/bill/118th-congress/h...  
1  https://www.congress.gov/bill/118th-congress/h...

In [32]:
df_bills.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10758 entries, 0 to 10757
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   bill_id           10758 non-null  int64 
 1   session_id        10758 non-null  int64 
 2   bill_number       10758 non-null  object
 3   status            10758 non-null  int64 
 4   status_desc       10758 non-null  object
 5   status_date       10758 non-null  object
 6   title             10758 non-null  object
 7   description       10758 non-null  object
 8   committee_id      10758 non-null  int64 
 9   committee         9425 non-null   object
 10  last_action_date  10758 non-null  object
 11  last_action       10758 non-null  object
 12  url               10758 non-null  object
 13  state_link        10758 non-null  object
dtypes: int64(4), object(10)
memory usage: 1.1+ MB


## Dataframe2: People

In [38]:
df_people = pd.read_csv(csv_people)

In [39]:
df_people.head(2)

people_id             name first_name middle_name  last_name suffix   
0       8955  Nydia Velazquez      Nydia          M.  Velazquez    NaN  \
1       8957    Frank Pallone      Frank         NaN    Pallone    Jr.   

  nickname  party_id party  role_id role district  followthemoney_eid   
0      NaN         1     D        1  Rep  HD-NY-7            17658223  \
1      NaN         1     D        1  Rep  HD-NJ-6            17658803   

   votesmart_id opensecrets_id      ballotpedia  knowwho_pid  committee_id  
0         26975      N00001102  Nydia_Velazquez       158981             0  
1         26951      N00000781    Frank_Pallone       158957             0

In [40]:
df_people.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 542 entries, 0 to 541
Data columns (total 18 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   people_id           542 non-null    int64 
 1   name                542 non-null    object
 2   first_name          542 non-null    object
 3   middle_name         183 non-null    object
 4   last_name           542 non-null    object
 5   suffix              21 non-null     object
 6   nickname            41 non-null     object
 7   party_id            542 non-null    int64 
 8   party               542 non-null    object
 9   role_id             542 non-null    int64 
 10  role                542 non-null    object
 11  district            542 non-null    object
 12  followthemoney_eid  542 non-null    int64 
 13  votesmart_id        542 non-null    int64 
 14  opensecrets_id      458 non-null    object
 15  ballotpedia         542 non-null    object
 16  knowwho_pid         542 no

##  Dataframe3: Sponsors

In [43]:
df_sponsors = pd.read_csv(csv_sponsors)

In [44]:
df_sponsors.head(2)

bill_id  people_id  position
0  1642307       9414         1
1  1642332       9414         1

## Dataframe4: History

In [53]:
df_history = pd.read_csv(csv_history)

In [54]:
df_history.head(2)

bill_id        date chamber  sequence   
0  1642307  2023-01-03  Senate         1  \
1  1642332  2023-01-03  Senate         1   

                                              action  
0  Submitted in the Senate, considered, and agree...  
1  Submitted in the Senate, considered, and agree...

## Investigation1: Interested in the actions but can't read the truncated output

In [55]:
pd.set_option('display.max_colwidth',None)

In [57]:
df_history.head(10)

bill_id        date chamber  sequence   
0  1642307  2023-01-03  Senate         1  \
1  1642332  2023-01-03  Senate         1   
2  1642332  2023-01-03  Senate         2   
3  1642361  2023-01-03  Senate         1   
4  1642361  2023-01-03  Senate         2   
5  1642389  2023-01-03  Senate         1   
6  1642421  2023-01-03  Senate         1   
7  1642453  2023-01-03  Senate         1   
8  1642480  2023-01-03  Senate         1   
9  1642512  2023-01-03  Senate         1   

                                                                                                                           action  
0  Submitted in the Senate, considered, and agreed to without amendment by Unanimous Consent. (consideration: CR S7; text: CR S7)  
1  Submitted in the Senate, considered, and agreed to without amendment by Unanimous Consent. (consideration: CR S7; text: CR S7)  
2                                                                                     Message on Senate action sent to the House.  
3  Submitted in the Senate, considered, and agreed to without amendment by Unanimous Consent. (consideration: CR S6; text: CR S6)  
4                                                                                     Message on Senate action sent to the House.  
5  Submitted in the Senate, considered, and agreed to without amendment by Unanimous Consent. (consideration: CR S7; text: CR S7)  
6                                      Submitted in the Senate. Placed on Senate Legislative Calendar under Over, Under the Rule.  
7  Submitted in the Senate, considered, and agreed to without amendment by Unanimous Consent. (consideration: CR S7; text: CR S7)  
8  Submitted in the Senate, considered, and agreed to without amendment by Unanimous Consent. (consideration: CR S6; text: CR S6)  
9  Submitted in the Senate, considered, and agreed to without amendment by Unanimous Consent. (consideration: CR S6; text: CR S6)

## Investigation2: Interested in the different types of actions. 1st look at any 'Unanimous' actions

In [58]:
df_history_1 = df_history[df_history['action'].str.contains('Unanimous')]

In [59]:
df_history.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28101 entries, 0 to 28100
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   bill_id   28101 non-null  int64 
 1   date      28101 non-null  object
 2   chamber   28101 non-null  object
 3   sequence  28101 non-null  int64 
 4   action    28101 non-null  object
dtypes: int64(2), object(3)
memory usage: 1.1+ MB


In [60]:
df_history_1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 355 entries, 0 to 27979
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   bill_id   355 non-null    int64 
 1   date      355 non-null    object
 2   chamber   355 non-null    object
 3   sequence  355 non-null    int64 
 4   action    355 non-null    object
dtypes: int64(2), object(3)
memory usage: 16.6+ KB


In [78]:
df_history['action'].nunique()

6227

## Investigation3: 355 actions containing 'Unanimous'. There are 6227 unique actions though, so no graphing yet

In [80]:
df_history['action'].value_counts()[:20]

action
Introduced in House                                                                        7093
Referred to the House Committee on Energy and Commerce.                                     811
Referred to the Subcommittee on Health.                                                     752
Referred to the House Committee on the Judiciary.                                           702
Referred to the House Committee on Education and the Workforce.                             535
Read twice and referred to the Committee on Health, Education, Labor, and Pensions.         486
Referred to the House Committee on Ways and Means.                                          471
Read twice and referred to the Committee on Finance.                                        417
Referred to the House Committee on Foreign Affairs.                                         410
Referred to the House Committee on Oversight and Accountability.                            393
Referred to the House Committee o

In [81]:
df_history_1['action'].nunique()

263

In [83]:
df_history_1['action'].value_counts()[:20]

action
Senate Committee on the Judiciary discharged by Unanimous Consent.                                                                                               20
Ordered to be Reported by Unanimous Consent.                                                                                                                     11
Ordered to be Reported (Amended) by Unanimous Consent.                                                                                                           10
Senate Committee on Environment and Public Works discharged by Unanimous Consent.                                                                                 5
Submitted in the Senate, considered, and agreed to without amendment by Unanimous Consent. (consideration: CR S7; text: CR S7)                                    4
Senate Committee on Health, Education, Labor, and Pensions discharged by Unanimous Consent.                                                                       4
Senate Co

## 263 unique actions that contain 'Unanimous' and only 3 have double digit occurrences. Also, they contain CR numbers, so this makes them unique